### PR TITLE
Security

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/interface/viewport.c [hunk 1]
```diff
@@ -2851,3 +2851,23 @@ void screen_get_map_xy_side_with_z(sint16 screenX, sint16 screenY, sint16 z, sin
...
+ *
+ * If an invalid rotation is detected and DEBUG_LEVEL_1 is enabled, an error
+ * will be reported.
+ *
+ * @returns rotation in range 0-3 (inclusive)
+ */
+int get_current_rotation()
+{
+    int rotation = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32);
+	int rotation_clamp = rotation & 3;
```
Comment (IntelOrca) [inline src/interface/viewport.c]: Not really a clamp, its a wrap.

Comment (janisozaur) [inline src/interface/viewport.c]: But int reality it is simply masked. anyway, fixed :)

Comment (IntelOrca) [inline src/interface/viewport.c]: Its not a clamp, its a wrap.

**Comment by IntelOrca** [comment]
Change it from ```rotation_clamp``` to ```rotation_wrap```

**Comment by IntelOrca** [comment]
What do the different debug levels mean? There should probably be an explanation of which debug level to use for something.

**Comment by janisozaur** [comment]
I renamed it to `_masked`, reflects reality better in my opinion, check the latest upload.

**Comment by janisozaur** [comment]
They don't have a specific meaning just yet, but I imagine you would use them to progressively provide more and more information about context you're in.

Like you could start logging memory contents at some deeper levels, when debugging some issues.

Is this fine with you?

**Comment by duncanspumpkin** [comment]
Are the diagnostic levels not the same as this https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/diagnostic.h

**Comment by janisozaur** [comment]
Not quite, `DIAGNOSTIC_LEVEL_*` are runtime and in discussion in another PR it was mentioned, this should be compile-time.

**Comment by IntelOrca** [comment]
@duncanspumpkin they are a bit mixed

	DIAGNOSTIC_LEVEL_FATAL (all levels)
	DIAGNOSTIC_LEVEL_ERROR (all levels)
	DIAGNOSTIC_LEVEL_WARNING (all levels)
	DIAGNOSTIC_LEVEL_VERBOSE (verbose level)
	DIAGNOSTIC_LEVEL_INFORMATION (?)

**Comment by janisozaur** [comment]
https://github.com/OpenRCT2/OpenRCT2/pull/2150#discussion_r43331582

**Comment by IntelOrca** [comment]
Something we could follow:
https://docs.asp.net/en/latest/fundamentals/logging.html

**Comment by IntelOrca** [comment]
We also need to define user logging and diagnostic logging. For example, if the level failed to load for such and such reason, we don't need to say what file and line it occurred on. If its a specific assertion, invalid value at a particular point then we do.

**Comment by janisozaur** [comment]
would using some library be feasible? http://stackoverflow.com/questions/613919/c-logging-libraries

**Comment by IntelOrca** [comment]
not another dependency!

**Comment by IntelOrca** [comment]
@janisozaur rebase this and I will merge this, the debug log can stay like this for the time being until we know what we want to do.

**Comment by janisozaur** [comment]
rebased

**Pull request merged**